### PR TITLE
gh-2640: Bump jackson version

### LINKED ISSUE #2640: Bump jackson-databind from 2.11.4 to 2.12.6.1
As done in koryphe: https://github.com/gchq/koryphe/pull/253

### PR DESCRIPTION
**Pull request by t92549**
# Related Issue

- Resolve #2640

### CODE REVIEW

**Comment by t92549** [comment]
This is currently failing due to koryphe 2.1.0 using a different jackson version, but that should be fixed when https://github.com/gchq/koryphe/pull/253 is merged

**Review by GCHQDeveloper314 — APPROVED** [review]
I've checked this locally and it works once Koryphe is updated. We shouldn't merge this PR until after the next release of Koryphe is out and the CI is passing here.

**Comment by t92549** [comment]
When https://github.com/gchq/Gaffer/pull/2644 is merged into `v2-alpha`, then this branch is updated, it should pass CI and can be merged.

**Pull request merged**
